### ISSUE TITLE
DatabaseClient select for single column returned as(String.class) question

### ISSUE THREAD

**Issue by RobMaskell** [spring-projects/spring-data-r2dbc#380]
I was playing with `DatabaseContext` in `spring-boot-starter-data-r2dbc:2.4.0-SNAPSHOT` and 
```
       return databaseClient.execute("select tablename from pg_tables")
            .as(String.class)
            .fetch().all()
            .flatMap(dropDbSql -> {
                System.out.println(dropDbSql);
                return Mono.empty();
            })
            .then();
```
return the exception
```
2020-06-04 21:31:40.971  WARN 76091 --- [  restartedMain] onfigReactiveWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'initializer' defined in class path resource [com/reactive/sample/cruddemo/config/DatabaseConfig.class]: Invocation of init method failed; nested exception is org.springframework.data.r2dbc.connectionfactory.init.UncategorizedScriptException: Failed to execute database script; nested exception is org.springframework.core.convert.ConversionFailedException: Failed to convert from type [io.r2dbc.postgresql.PostgresqlRow] to type [java.lang.String] for value 'PostgresqlRow{context=ConnectionContext{client=io.r2dbc.postgresql.client.ReactorNettyClient@371e399d, codecs=io.r2dbc.postgresql.codec.DefaultCodecs@2df0c20b, connection=PostgresqlConnection{client=io.r2dbc.postgresql.client.ReactorNettyClient@371e399d, codecs=io.r2dbc.postgresql.codec.DefaultCodecs@2df0c20b}, configuration=PostgresqlConnectionConfiguration{applicationName='r2dbc-postgresql', autodetectExtensions='true', connectTimeout=null, database='postgres', extensions=[], fetchSize=io.r2dbc.postgresql.PostgresqlConnectionConfiguration$Builder$$Lambda$636/0x000000084053b440@39e687d, forceBinary='false', host='localhost', options='null', password='********', port=5432, schema='null', username='postgres'}, portalNameSupplier=io.r2dbc.postgresql.DefaultPortalNameSupplier@4f03ba04, statementCache=IndefiniteStatementCache{cache={}, client=io.r2dbc.postgresql.client.ReactorNettyClient@371e399d, counter=0}}, columns=[Field{column=2, dataType=19, dataTypeModifier=-1, dataTypeSize=64, format=FORMAT_TEXT, name='tablename', table=11621}], isReleased=false}'; nested exception is java.lang.IllegalArgumentException: Cannot decode value of type java.lang.String
```
but this
```
    return databaseClient.execute("select '' || tablename from pg_tables")
            .as(String.class)
            .fetch().all()
            .flatMap(dropDbSql -> {
                System.out.println(dropDbSql);
                return Mono.empty();
            })
            .then();
```
works fine, just wondered if that was the expected behaviour?

**Comment by mp911de** [comment]
Can you provide a bit more context? We have a [few tests](https://github.com/spring-projects/spring-data-r2dbc/blob/be00ea84effbfd5b002cad225a7f562b8efbd52d/src/test/java/org/springframework/data/r2dbc/core/AbstractDatabaseClientIntegrationTests.java#L393-L407) that make sure single-column projections work properly.

**Comment by RobMaskell** [comment]
Code is from a `DatabasePopulator` I was writing that drops all the tables in a schema, specifically
```
    @Override
    public Mono<Void> populate(Connection connection) throws ScriptException {
```
Had a look at the types and it seems the `tablename` column is of type `name` perhaps there isn't a mapping for that to `String.class` for the postgres dialect?

**Comment by RobMaskell** [comment]
Hmmm as far as I can tell `Name` is there https://github.com/r2dbc/r2dbc-postgresql/blob/master/src/main/java/io/r2dbc/postgresql/codec/StringCodec.java

**Comment by mp911de** [comment]
I wasn't able to reproduce the issue. I initially anticipated some changes in the classloader that the registered converters are no longer applicable because of the `Row` class identity but that doesn't seem to be the case. Can you provide a minimal reproducer?

**Comment by RobMaskell** [comment]
see attached, run in vscode, openjdk 11.0.7
[repro.zip](https://github.com/spring-projects/spring-data-r2dbc/files/4748215/repro.zip)

**Comment by mp911de** [comment]
Thanks a lot. The issue is that the Postgres driver as of 0.8.2.RELEASE does not support decoding of `NAME` typed columns. The support was added only with 0.8.3.RELEASE. I got sidetracked by converters and didn't recognize that `Name` type support was added recently.

So upgrading to Arabba-SR4 (or directly 0.8.3.RELEASE for r2dbc-postgresql) will fix the issue.

Closing this ticket as that's a driver concern.
